src/Client/StorePaymentMethodOnChain.php

```diff
@@ -0,0 +1,139 @@
+ */
+class StorePaymentMethodOnChain extends AbstractStorePaymentMethodClient
+{
+    public function getPaymentMethods(string $storeId): array
+    {
+        $url = $this->getApiUrl() . 'stores/' . urlencode($storeId) . '/payment-methods/OnChain';
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+        $response = CurlClient::request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            $r = [];
+            $data = json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            foreach ($data as $item) {
+                $r[] = new \BTCPayServer\Result\StorePaymentMethodOnChain($item, $item['cryptoCode']);
+            }
+            return $r;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getPaymentMethod(string $storeId, string $cryptoCode): \BTCPayServer\Result\StorePaymentMethodOnChain
+    {
+        $url = $this->getApiUrl() . 'stores/' . urlencode($storeId) . '/payment-methods/OnChain/' . $cryptoCode;
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+        $response = CurlClient::request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            $data = json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            return new \BTCPayServer\Result\StorePaymentMethodOnChain($data, $data['cryptoCode']);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+     * Update a payment method.
+     *
+     * @param string $storeId
+     * @param string $cryptoCode
+     *
+     * @param array $settings
+     *  Array of data to update. e.g.
+     *    [
+     *      'enabled' => true,
+     *      'derivationScheme' => 'xpub...',
+     *      'label' => 'string',
+     *      'accountKeyPath' => "abcd82a1/84'/0'/0'"
+     *    ]
+     *
+     * @return \BTCPayServer\Result\StorePaymentMethodOnChain
+     * @throws \JsonException
+     *
+     */
+    public function updatePaymentMethod(string $storeId, string $cryptoCode, array $settings): \BTCPayServer\Result\StorePaymentMethodOnChain
+    {
+        $url = $this->getApiUrl() . 'stores/' . urlencode($storeId) . '/payment-methods/OnChain/' . $cryptoCode;
+        $headers = $this->getRequestHeaders();
+        $method = 'PUT';
+        $response = CurlClient::request($method, $url, $headers, json_encode($settings));
+
+        if ($response->getStatus() === 200) {
+            $data = json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            return new \BTCPayServer\Result\StorePaymentMethodOnChain($data, $data['cryptoCode']);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function previewPaymentMethodAddresses(string $storeId, string $cryptoCode): array
```

Comment:
We don't have a class for this type of response?

Reply:
Not yet, we will need some OnChainAddress class or something like that.

Reply:
DO you want to include this in this PR or at a later date?

Reply:
added classes